deps/streamsearch/sbmh.js

```diff
@@ -168,15 +169,15 @@ SBMH.prototype._sbmh_feed = function (data) {
 
   // Lookbehind buffer is now empty. We only need to check if the
   // needle is in the haystack.
-  pos = data.indexOf(needle, pos + ((pos >= 0) * this._bufpos))
+  pos = data.indexOf(needle, pos + this._bufpos)
 
   if (pos !== -1) {
     ++this.matches
-    if (pos > 0) { this.emit('info', true, data, this._bufpos, pos) } else { this.emit('info', true) }
+    if (pos === 0) { this.emit('info', true) } else { this.emit('info', true, data, this._bufpos, pos) }
     return (this._bufpos = pos + needleLength)
   }
 
-  pos = len - needleLength
+  pos = Math.max(len - needleLastCharIndex, 0)
```

Comment:
```suggestion
  pos = len - needleLastCharIndex
  pos = pos < 0 ? 0
```
`Math.max` is unfortunately slower than if statement